docs/secret_management.md

```diff
@@ -0,0 +1,172 @@
+# Secrets Management for the API and APP keys
+
+The Datadog-Operator is able to retrieves the datadog credentials securely thanks to 3 different methods.
+
+## 1. Plain credentials in DatadogAgent resource
+
+this is the simplest way to provide the credentials that will be used by the agents.
+
+```yaml
+apiVersion: datadoghq.com/v1alpha1
+kind: DatadogAgent
+metadata:
+  name: datadog
+spec:
+  credentials:
+    apiKey: "<api-key>"
+    appKey: "<app-key>"
+  # ...
+```
+
+The credentials provided here will be stored in a Secret created by the Operator. By setting properly the `RBAC` on the `DatadogAgent` CRD. It is possible to limit who is able to see those credentials.
+But still, it is not the best solution in terms of security. This solution is good for testing purposes.
```

Comment:
If this isn't secure, we should remove this whole section and not even offer it as an option

Reply:
if someone want to test the operator quickly, this is the simpler way. That is why I'm adding this option in the doc.

maybe I can move this sentence at the beginning of this section, wdyt?

Reply:
Yes, I'd suggest to remove it and instead add the following to line 7:
"This is the simplest way to provide the credentials to the agents. This method is recommended for testing purposes only."

Reply:
PR updated.
I think we are good to merge it.